src/vs/workbench/contrib/scm/browser/dirtydiffDecorator.ts

```diff
@@ -1101,25 +1085,26 @@ export async function getOriginalResource(quickDiffService: IQuickDiffService, u
 	return quickDiffs.length > 0 ? quickDiffs[0].originalResource : null;
 }
 
+type LabeledChange = { change: IChange; labels: string[]; uri: URI };
+
 export class DirtyDiffModel extends Disposable {
 
-	private _originalResource: URI | null = null;
-	private _originalModel: IResolvedTextEditorModel | null = null;
+	private _quickDiffs: QuickDiff[] | null = null;
+	private _originalModels: Map<string, IResolvedTextEditorModel> = new Map(); // key is uri.toString()
 	private _model: ITextFileEditorModel;
-	get original(): ITextModel | null { return this._originalModel?.textEditorModel || null; }
-	get modified(): ITextModel | null { return this._model.textEditorModel || null; }
+	get original(): ITextModel[] { return Array.from(this._originalModels.values()).map(original => original.textEditorModel); }
```

Comment:
Should this be cached? Every time the prop get accessed, two arrays are being created.

Reply:
Added caching.